synapse_invite_checker/invite_checker.py

```diff
@@ -130,15 +130,21 @@ def __init__(self, config: InviteCheckerConfig):
 
         self.url = urlparse(config.federation_list_url)
 
-        # if no certificate is specified, we assume the connection uses http and no MTLS
+        # Handle different cases based on URL scheme and mTLS requirements
         client_cert = None
         if config.federation_list_client_cert:
+            # If a certificate is provided, always use it
             with open(config.federation_list_client_cert) as file:
                 content = file.read()
 
             client_cert = PrivateCertificate.loadPEM(content)
-        elif self.url.scheme != "http":
-            msg = "No mtls cert and scheme is not http"
+        elif self.url.scheme == "https" and config.federation_list_require_mtls:
+            # HTTPS with required mTLS but no certificate
+            msg = "No mtls cert and scheme is https with mTLS required"
+            raise Exception(msg)
+        elif self.url.scheme != "http" and self.url.scheme != "https":
+            # Neither HTTP nor HTTPS
+            msg = "URL scheme must be either http or https"
```

Comment:
Can you add a test for this exception? Otherwise jason will yell at me again for not being strict enough :)